src/audio/Kconfig

```diff
@@ -38,6 +38,23 @@ config COMP_VOLUME_WINDOWS_FADE
 	 SOF_VOLUME_WINDOWS_FADE for the volume instance to use
 	 this ramp shape.
 
+config COMP_VOLUME_LINEAR_RAMP
+       bool "Linear ramp volume transitions support"
+	   default y
+	   depends on IPC_MAJOR_3
+       help
+         This option enables volume linear ramp shape.
+
+config COMP_VOLUME_Q1_23
+	bool
+	default y
+	depends on IPC_MAJOR_4
+
+config COMP_VOLUME_Q8_16
+	bool
+	default y
+	depends on IPC_MAJOR_3
+
```

Comment:
it somehow feels like all of the above should be simple macros instead of config options.

Reply:
The best would be to add select in COMP_VOLUME kconfig
select A if IPC4
select B if IPC3 
but it didnt work so I ended with a standard config, just hidden

@singalsu @ranj063 where do you want to put this?

Reply:
@abonislawski how about hiding this as ops or fields in the volume specific dat. struct vol_data? You can have fields such as q_notation that are set to Q1.23 and Q1.31 depending on the value and ops for doing IPC specific operations that will be set only where relevant

Reply:
Not sure if this will be optimal solution to use AE_SRAI64 with shifts stored in variables.
@singalsu already mentioned to hard code and optimize all shifts to minimum.

Reply:
@abonislawski not following what you mean by not optimal? You can still hardcode whatever you want based on the IPC version. Just wont be part of the config and also it will reduce al the #if/#else;s

Reply:
@abonislawski fwiw, we had a recent bug with a similar Kconfig dependency.  If the volume type is only used in 1 module then we could just do a 
```c
#if CONFIG_IPC3
#define COMP_VOLUME_Q8_16 
#endif
```
and so on. This keeps some simplicity in the Kconfig and the format localized to volume.

Reply:
Ok, moved to volume.h